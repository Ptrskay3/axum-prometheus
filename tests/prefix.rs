mod common;
use common::{echo, BoxBody};

use http::Request;
use tower::{Service, ServiceBuilder, ServiceExt};

#[tokio::test]
async fn metric_handle_rendered_correctly_with_prefix() {
    let (layer, handle) = axum_prometheus::PrometheusMetricLayerBuilder::new()
        .with_prefix("pref")
        .with_default_metrics()
        .build_pair();

    let mut service = ServiceBuilder::new().layer(layer).service_fn(echo);

    let req = Request::builder().body(BoxBody::default()).unwrap();
    let _res = service.ready().await.unwrap().call(req).await.unwrap();
    insta::with_settings!({
            filters =>
            vec![
                (
                    r"\b[-+]?[0-9]*\.?[0-9]+\b\\npref_http_requests_duration_seconds_count",
                    "",
                )
            ]
        },
    {

        insta::assert_yaml_snapshot!(handle.render());
    }
    );
}
