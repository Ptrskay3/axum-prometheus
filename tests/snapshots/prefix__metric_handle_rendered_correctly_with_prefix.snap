---
source: tests/prefix.rs
expression: handle.render()
---
"# TYPE pref_http_requests_total counter\npref_http_requests_total{method=\"GET\",status=\"200\",endpoint=\"/\"} 1\n\n# TYPE pref_http_requests_pending gauge\npref_http_requests_pending{method=\"GET\",endpoint=\"/\"} 0\n\n# TYPE pref_http_requests_duration_seconds histogram\npref_http_requests_duration_seconds_bucket{method=\"GET\",status=\"200\",endpoint=\"/\",le=\"0.005\"} 1\npref_http_requests_duration_seconds_bucket{method=\"GET\",status=\"200\",endpoint=\"/\",le=\"0.01\"} 1\npref_http_requests_duration_seconds_bucket{method=\"GET\",status=\"200\",endpoint=\"/\",le=\"0.025\"} 1\npref_http_requests_duration_seconds_bucket{method=\"GET\",status=\"200\",endpoint=\"/\",le=\"0.05\"} 1\npref_http_requests_duration_seconds_bucket{method=\"GET\",status=\"200\",endpoint=\"/\",le=\"0.1\"} 1\npref_http_requests_duration_seconds_bucket{method=\"GET\",status=\"200\",endpoint=\"/\",le=\"0.25\"} 1\npref_http_requests_duration_seconds_bucket{method=\"GET\",status=\"200\",endpoint=\"/\",le=\"0.5\"} 1\npref_http_requests_duration_seconds_bucket{method=\"GET\",status=\"200\",endpoint=\"/\",le=\"1\"} 1\npref_http_requests_duration_seconds_bucket{method=\"GET\",status=\"200\",endpoint=\"/\",le=\"2.5\"} 1\npref_http_requests_duration_seconds_bucket{method=\"GET\",status=\"200\",endpoint=\"/\",le=\"5\"} 1\npref_http_requests_duration_seconds_bucket{method=\"GET\",status=\"200\",endpoint=\"/\",le=\"10\"} 1\npref_http_requests_duration_seconds_bucket{method=\"GET\",status=\"200\",endpoint=\"/\",le=\"+Inf\"} 1\npref_http_requests_duration_seconds_sum{method=\"GET\",status=\"200\",endpoint=\"/\"} {method=\"GET\",status=\"200\",endpoint=\"/\"} 1\n\n"

